### PR TITLE
Linting: Update instructions for ESLint v9

### LINKED ISSUE #27818: Linting: Tutorial to setup eslint with vscode not working anymore.
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

The tutorial for setting up eslint for vscode doesn't work, it's because eslint uses a new configuration file as default, but the extension expects the old one, so the fix for this is to switch to  "Pre-release version" on extension, which took me a long time to figure out. 

I also think that the setup used in the tutorial is not done in the same way as in the getting started section in the eslint website, but i'm not sure if this is a significant issue. 

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-linting

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by gingkapls**
## Because
The recent ESLint V9 update made breaking changes to the default configuration file format, which resulted in the old instructions becoming out-of-date and some plugins to stop working with the new default format among other changes.

## This PR
- Corrects link text from eslint to ESLint
- Removes DigitalOcean tutorial for setting up ESLint extension as it's outdated and not needed anymore
- Adds a lesson note about the current version of ESLint not picking up configurations for workspace sub-directories and fixing that behaviour
- Removes instructions for making ESLint and Prettier work together, as they work out of the box now

## Issue

Closes #27818 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by MaoShizhong** [comment]
@gingkapls You can ignore the assignment linting error for the moment.

Putting this on hold while investigating further things regarding the ESLint extension, and flat config rules, as it seems we might need to tweak some stuff due to confusion new systems.

**Comment by gingkapls** [comment]
@MaoShizhong [eslint-plugin-react does not work with eslint v9 yet](https://github.com/jsx-eslint/eslint-plugin-react/issues/3699), and since Vite pulls it in as a dependency, it too defaults to using ESLint v8 and the older `.eslintrc.cjs` format.

This creates a discrepancy between the instructions in the Linting lesson and the ones in [Setting Up a React Environment](https://www.theodinproject.com/lessons/node-path-react-new-setting-up-a-react-environment#keeping-it-clean) where we suggest using `eslint-config-prettier`. 

I think the instructions in the React lesson will have to remain the same until Vite updates to ESLint v9, but I haven't reached the React section of the curriculum myself so I cannot give any further input on the impact of the discrepancy between the two lessons.

**Comment by MaoShizhong** [comment]
@gingkapls Thanks for looking into that. I just came across the same thing regarding some non-curriculum options, which will be resolved once the necessary plugins support v9+.

For now, I'm not concerned about the React part. The linting lesson is about introducing Linters and getting started with ESLint. In the React portion, all the necessary deps and files are generated when you run the Vite CLI command, so it does everything for you. The only difference will be the eslint config won't use the flat config format (which will be fine pre-v9). Vite will update their create CLI tool for the React template once they are able to (which will likely be when `eslint-plugin.

We actually do not need that `eslint-config-prettier` bit, to be totally honest. All it does is disable the following rules: https://github.com/prettier/eslint-config-prettier/blob/main/index.js none of which are activated in any of the plugins that come with the Vite template:
```
    'eslint:recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
```

So that part can actually be removed, but the rest of the React set up lesson can remain as is.

File: javascript/javascript_in_the_real_world/linting.md
Comment (MaoShizhong) [inline javascript/javascript_in_the_real_world/linting.md:37]:
```suggestion
The current version of the extension (2.4.4) will only pick up the workspace folder's ESLint config file, and not a config file for a subdirectory of that workspace. Switching to the pre-release version solves this. You will also need to enable `Eslint: Use Flat Config` in VSCode's settings.
```

File: javascript/javascript_in_the_real_world/linting.md
Comment (MaoShizhong) [inline javascript/javascript_in_the_real_world/linting.md:31]: ### DO NOT COMMIT THIS SUGGESTION

```suggestion
   - DO NOT COMMIT THIS. THIS IS JUST TO SHOW THE POSITION OF THE SUGGESTION
1. There is an [ESLint extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) with which you can get automatic lint highlighting for your files as you write, without you needing to rerun the `eslint` command every time. If your project also contains an ESLint configuration file, the extension will automatically use those rules for that project.
```

I think it would be beneficial to add a small sub-bullet linking to the ["Configure ESLint" page](https://eslint.org/docs/latest/use/configure/) and potentially even adding a note about [ignoring files](https://eslint.org/docs/latest/use/configure/ignore) (regarding the difference between `ignores` being a lone object property vs in a config object with multiple).

Tbh I'm not 100% sure about including the `ignores` thing, or at least the best way to not be more confusing than not. We should definitely link to the config overview page so people are aware of where they can go to learn about amending the ESLint config.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]

File: javascript/javascript_in_the_real_world/linting.md
Comment (gingkapls) [reply javascript/javascript_in_the_real_world/linting.md:31]: Hmm, I see the merit in adding a sub-bullet for the configuration page.

As for the part about ignoring files, I'm not sure either.
You know better about the later parts of the curriculum, so if you feel that knowing about this peculiar behaviour of `ignores` is something that may help learners in the future projects/assignments, then we can go ahead and add it.

If it's something that doesn't see much use within the curriculum, but is common in non-curriculum environment, then perhaps we could trust the learners to look for it, since the top result for `eslint global ignore` is the `Ignore Files` page in ESLint docs.


> In your eslint.config.js file, if an ignores key is used without any other keys in the configuration object, then the patterns act as global ignores

The relevant reading on `ignores` special behaviour is just a single line, so I don't think we'd go wrong either way. I'm fine with whatever your call is.

Comment (MaoShizhong) [reply javascript/javascript_in_the_real_world/linting.md:31]: Let's omit it for now and see how it goes. I'm going to be PRing an improvement to ESLint docs if there isn't already one open, so hopefully it'll be improved soon enough, be that from my changes or someone elses.

Comment (gingkapls) [reply javascript/javascript_in_the_real_world/linting.md:33]: Do the changes look good?

Comment (MaoShizhong) [reply javascript/javascript_in_the_real_world/linting.md:33]: > Do the changes look good?

I'll have a proper review in a bit. For future reference, if changes are requested and you're now in a state where it's ready for re-review, you can re-request a review from that maintainer by hitting the rerequest button in the top right:
image

Comment (gingkapls) [reply javascript/javascript_in_the_real_world/linting.md:33]: I clicked the button :D

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
I think everything else should be good to go, just the one small "might as well" below. Thank you for looking into and working on this as well. Not easy when the tool's release itself wasn't the cleanest...

**Comment by gingkapls** [comment]
It was a fun experience, and I got to help out too. Thanks for helping and guiding me throughout all this :)

**Pull request merged**
